crates/sui-adapter/src/adapter.rs

```diff
@@ -234,7 +234,7 @@ pub fn publish<
     ctx: &mut TxContext,
     gas_status: &mut SuiGasStatus,
 ) -> Result<(), ExecutionError> {
-    gas_status.charge_publish_package(module_bytes.iter().map(|v| v.len()).sum())?;
+    gas_status.charge_publish_package()?;
```

Comment:
So this changes the module publish cost from per-bytes to constant. What's the rational behind this?

Reply:
@lxfind this was discussed and agreed upon in our meetings. The rationale was that publish & verif is tricky to price, and that size was not a good indicator of complexity.
Now I'm more than happy to make it a linear function which is what I've always wanted: `flat_verif_cost` + `size_dependent_publish_cost` * `package_size`
This also keeps backward compat with old method if we want

Reply:
Indeed size is not a good indicator of complexity and we currently don't have a better way to track complexity. I am still curious about the trade-off between a flat fee vs size-based fee: size-based fee certainly is still more representative than flat? Is the intention to simplify user experience here?

Reply:
@lxfind I added a trait which attempts to give deterministic size.
The plan is to make this an auto derived trait with options which can also enforce that all struct members have a deterministic size method.
I think this is good for now while we flesh out something better,